src/components/feed/upload/Category/TagSelector/TagSelectOptions.tsx

```diff
@@ -36,6 +38,27 @@ export default function TagSelectOptions({ onClose, onSave, feedData }: TagSelec
         category.id === feedData.mainCategoryId && category.children.some((tag) => tag.id === feedData.categoryId),
     ) ?? null;
 
+  const { data: myProfile } = useGetMemberProfileOfMe();
+
+  const sortedSoptActivities = useMemo(() => {
+    if (!myProfile?.soptActivities) {
+      return [];
+    }
+    const sorted = [...myProfile.soptActivities];
+    sorted.sort((a, b) => b.generation - a.generation);
+    return sorted;
+  }, [myProfile?.soptActivities]);
+
+  const partCategory =
+    parentCategory && parentCategory.children.find((category) => category.name === sortedSoptActivities[0]?.part);
```

Comment:
```suggestion
  const partCategory =
    parentCategory?.children.find((category) => category.name === sortedSoptActivities[0]?.part);
```

p5;
별 거 아니지만 이렇게 쓰면 더 간단할 것 같아요!